src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+import { Error } from '../../types/Error';
+
+type Props = {
+  todos: Todo[],
+  setErrorType: (error: Error) => void,
+};
+
+export const TodoList: React.FC<Props> = ({ todos, setErrorType }) => {
+  return (
+    <section className="todoapp__main" data-cy="TodoList">
+      {todos.map(todo => {
+        const { title, id, completed } = todo;
+
+        return (
+          <div
+            data-cy="Todo"
+            className={cn('todo', { completed })}
+            key={id}
+          >
+            <label className="todo__status-label">
+              <input
+                data-cy="TodoStatus"
+                type="checkbox"
+                className="todo__status"
+              />
+            </label>
+
+            <span data-cy="TodoTitle" className="todo__title">
+              {title}
+            </span>
+
+            {/* Remove button appears only on hover */}
+            <button
+              type="button"
+              className="todo__remove"
+              data-cy="TodoDelete"
+              onClick={() => setErrorType(Error.Delete)}
+            >
+              ×
+            </button>
+
+            {/* overlay will cover the todo while it is being updated */}
```

Comment:
Do not leave unnecessary comments.

Reply:
Not fixed 😢

Reply:
Sorry, I forgot about another components. Now all will be fine )